File: 2026-01/src/day_31/boj_6603/Main.java
package day_31.boj_6603;

import java.io.*;
import java.util.*;

public class Main {

    static StringBuilder sb = new StringBuilder();
    static int k;
    static int[] arr;
    static int[] sel;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        while (true) {
            st = new StringTokenizer(br.readLine());
            k = Integer.parseInt(st.nextToken());
            if (k == 0) break;

            arr = new int[k];
            for (int i = 0; i < k; i++) {
                arr[i] = Integer.parseInt(st.nextToken());
            }
            sel = new int[6];

            dfs(0, 0);

            sb.append("\n");
        }

        System.out.println(sb);
    }

    static void dfs(int idx, int selIdx) {
        if (selIdx == 6) {
            for (int x : sel) {
                sb.append(x).append(" ");
            }
            sb.append("\n");
            return;
        }

        for (int i = idx; i < k; i++) {
            sel[selIdx] = arr[i];
            dfs(i + 1, selIdx + 1);
        }
    }
}

File: 2026-01/src/day_31/boj_6603/Main.cpp
#include <bits/stdc++.h>
using namespace std;

int k;
int arr[13];
int sel[6];

void dfs(int idx, int sidx) {
    if (sidx == 6) {
        for (int x : sel) {
            cout << x << ' ';
        }
        cout << '\n';
        return;
    }

    for (int i = idx; i < k; i++) {
        sel[sidx] = arr[i];
        dfs(i + 1, sidx + 1);
    }
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    while (true) {
        cin >> k;
        if (k == 0) break;

        for (int i = 0; i < k; i++) cin >> arr[i];

        dfs(0, 0);
        cout << '\n';
    }
}
